identity_stardust/src/client/client_ext.rs

```diff
@@ -0,0 +1,616 @@
+
+    let document = client.publish_did_output(&secret_manager, output).await.unwrap();
+
+    let resolved = client.resolve_did(document.id()).await.unwrap();
+
+    assert_eq!(document, resolved);
+  }
+
+  #[tokio::test]
+  async fn test_client_publish_update() {
+    let client: Client = client();
+    let (address, secret_manager) = get_address_with_funds(&client).await;
+    let initial_document = generate_document(&valid_did());
+    let rent_structure: RentStructure = client.get_rent_structure().await.unwrap();
+
+    let output = client
+      .new_did(address, initial_document, Some(rent_structure.clone()))
+      .await
+      .unwrap();
+
+    let mut document = client.publish_did_output(&secret_manager, output).await.unwrap();
+
+    let method_url = document
+      .resolve_method("#key-1", Some(MethodScope::VerificationMethod))
+      .unwrap()
+      .id()
+      .to_owned();
+
+    document
+      .attach_method_relationship(
+        &method_url,
+        identity_did::verification::MethodRelationship::Authentication,
+      )
+      .unwrap();
+
+    let alias_output: AliasOutput = client.update_did(document).await.unwrap();
+
+    let rent_structure = client.get_rent_structure().await.unwrap();
+
+    let alias_output: AliasOutput = AliasOutputBuilder::from(&alias_output)
+      .with_minimum_storage_deposit(rent_structure)
+      .finish()
+      .unwrap();
+
+    let document: StardustDocument = client.publish_did_output(&secret_manager, alias_output).await.unwrap();
+
+    let resolved = client.resolve_did(document.id()).await.unwrap();
+
+    assert_eq!(document, resolved);
+  }
+
+  #[tokio::test]
+  async fn test_client_delete() {
+    let client: Client = client();
+    let network_hrp: String = get_network_hrp(&client).await.unwrap();
+
+    let (address, secret_manager) = get_address_with_funds(&client).await;
+    let address_bech32: String = address.to_bech32(network_hrp);
+    let initial_balance: u64 = get_address_balance(&client, &address_bech32).await;
+
+    let initial_document = generate_document(&valid_did());
+
+    let output = client.new_did(address, initial_document, None).await.unwrap();
+
+    let document: StardustDocument = client.publish_did_output(&secret_manager, output).await.unwrap();
+
+    client
+      .delete_did_output(&secret_manager, address, document.id())
+      .await
+      .unwrap();
+
+    client.consolidate_funds(&secret_manager, 0, 0..1).await.unwrap();
+
+    // It takes time for the deletion to propagate.
+    tokio::time::sleep(std::time::Duration::from_secs(10)).await;
+
+    let error = client.resolve_did(document.id()).await.unwrap_err();
+
+    assert!(matches!(error, Error::ResolutionError(iota_client::Error::NotFound)));
+
+    let balance: u64 = get_address_balance(&client, &address_bech32).await;
+
+    assert_eq!(initial_balance, balance);
+  }
+
+  #[tokio::test]
+  async fn test_client_deactivate_and_reactivate() {
+    let client: Client = client();
```

Comment:
We usually don't include unit tests that perform the same actions as the examples, since they're mostly redundant and interacting with the Tangle is slow.

The Stardust examples should be hooked up in CI at this point I guess.